linux_os/guide/system/accounts/accounts-pam/display_login_attempts/ansible/shared.yml

```diff
@@ -4,10 +4,11 @@
 # complexity = low
 # disruption = low
 
-{{% if product in ["sle12", "sle15"] or "ubuntu" in product %}}
-{{{ ansible_ensure_pam_module_configuration('/etc/pam.d/login', 'session', 'required', 'pam_lastlog.so', 'showfailed', '', 'BOF') }}}
-{{{ ansible_remove_pam_module_option_configuration('/etc/pam.d/login', 'session', '', 'pam_lastlog.so', 'silent') }}}
+{{% if product in ["sle12", "sle15"] or 'ubuntu' in product %}}
+{{% set pam_lastlog_path = "/etc/pam.d/login" %}}
 {{% else %}}
-{{{ ansible_ensure_pam_module_configuration('/etc/pam.d/postlogin', 'session', 'required', 'pam_lastlog.so', 'showfailed', '', 'BOF') }}}
-{{{ ansible_remove_pam_module_option_configuration('/etc/pam.d/postlogin', 'session', '', 'pam_lastlog.so', 'silent') }}}
+{{% set pam_lastlog_path = "/etc/pam.d/postlogin" %}}
 {{% endif %}}
+
+{{{ ansible_ensure_pam_module_configuration(pam_lastlog_path, 'session', '[default=1]', 'pam_lastlog.so', 'showfailed', '', 'BOF') }}}
```

Comment:
This seem to fail to insert the showfailed line in case I have a commented out line in my config like:
```
#%PAM-1.0
auth      requisite  pam_nologin.so
auth      include    common-auth
account   include    common-account
password  include    common-password
session   required   pam_loginuid.so
session   optional   pam_keyinit.so force revoke
session   include    common-session
#session  optional   pam_lastlog.so nowtmp showfailed
session   optional   pam_mail.so standard
auth    required    pam_tally2.so deny=3 onerr=fail
```

Reply:
@teacup-on-rockingchair , I tried to apply the remediation in a file with this content and the line was inserted at the beginning of the file (BOF):
```
session    [default=1]    pam_lastlog.so
#%PAM-1.0
auth      requisite  pam_nologin.so
auth      include    common-auth
account   include    common-account
password  include    common-password
session   required   pam_loginuid.so
session   optional   pam_keyinit.so force revoke
session   include    common-session
#session  optional   pam_lastlog.so nowtmp showfailed
session   optional   pam_mail.so standard
auth    required    pam_tally2.so deny=3 onerr=fail
```
I see the line used for remediation and even the `after_match` regex passed to the macro might be different among distros. I can create a jinja2 condition in these remediation and make it more flexible to be maintained.

Reply:
Thanks @teacup-on-rockingchair . Investigating this case I noticed the remediation can be improved a little more.

Reply:
It should be fine now @teacup-on-rockingchair